### ISSUE TITLE
Tectonic installer stucks at "Starting Tectonic" step.

### ISSUE THREAD

**Issue by megastallman** [coreos/tectonic-installer#2251]
## What keywords did you search in tectonic-installer issues before filing this one?
tectonic installer

## Is this a BUG REPORT or FEATURE REQUEST?
BUG REPORT

## Versions

* Tectonic version: 1.7.5-tectonic.1
* Platform: metal

## What happened?
Tectonic installer stucks at "Starting Tectonic" step.

## What you expected to happen?
It should have run successfully or break with an error. It should also have more verbosity.

## How to reproduce it (as minimally and precisely as possible)?
Run through the Tectonic Baremetal installation article. Press the "Apply" button, see the "Poweron" box, Power your servers on and check the PXE boot for the first time.
During the "Starting Tectonic", wait for more than an hour, to make sure it's stuck forever.
Log in to any node and watch the following message 
`Oct 27 16:30:32 node1.ololo.com kubelet-wrapper[1717]: E1027 16:30:32.124704    1717 reflector.go:190] k8s.io/kubernetes/pkg/kubelet/kubelet.go:400: Failed to list *v1.Service: Get https://controllers.ololo.com:443/api/v1/services?resourceVersion=0: dial tcp 10.10.9.97:443: getsockopt: connection refused
Oct 27 16:30:32 node1.ololo.com kubelet-wrapper[1717]: E1027 16:30:32.323950    1717 reflector.go:190] k8s.io/kubernetes/pkg/kubelet/kubelet.go:408: Failed to list *v1.Node: Get https://controllers.ololo.com:443/api/v1/nodes?fieldSelector=metadata.name%3Dnode1.ololo.com&resourceVersion=0: dial tcp 10.10.9.97:443: getsockopt: connection refused`
with `journalctl -f`
I've got the same result a couple of times.

## Anything else we need to know?
Our servers are being wiped manually with "# dd if=/dev/zero of=/dev/sda" before the reinstallation to avoid any possible post-effects. Just in case.

I've figured out, that there is a service(on node1), called "tectonic", which is actually the installer, but its status says, that it had failed due to missing dependencies. So, is this service actually the one to install these dependencies? Or are they circular?
I ran that service manually and saw the installation progress finished. Now the cluster works, but anyway, this should be either fixed, or supervised in case of such silent failures. These failures should never be silent. The web-interface operator should see what's happening.

**Comment by fforootd** [comment]
This looks like a duplicate of #2129

**Comment by megastallman** [comment]
No, it doesn't look like #2129. I've tried all the advice posted there. Nothing helped. And I didn't have those errors.
Only the # systemctl start tectonic command helped.

**Comment by mxinden** [comment]
@megastallman Thanks for reporting this issue. Yes, the `tectonic` systemd unit does parts of the cluster installation after bootkube finished. Could you  post the logs of the stuck `tectonic` and finished `bootkube` systemd unit from node1 here?

**Comment by megastallman** [comment]
Hi @mxinden! Thanks for reply. Here is my "tectonic" service [log](https://github.com/coreos/tectonic-installer/files/1433911/tectonic.log), and its status now is:
`
node1 core # systemctl status tectonic
● tectonic.service - Bootstrap a Tectonic cluster                                                                                                                       
   Loaded: loaded (/etc/systemd/system/tectonic.service; disabled; vendor preset: disabled)                                                                             
   Active: inactive (dead)                                                                                                                                              
Condition: start condition failed at Wed 2017-11-01 06:21:34 UTC; 4h 48min ago                                                                                          
           └─ ConditionPathExists=!/opt/tectonic/init_tectonic.done was not met
`
While the cluster works fine after that forced installation.

**Comment by mxinden** [comment]
Hi @megastallman,

 The *tectonic* logs you uploaded look fine, they finished with `Tectonic installation is done`. In addition the `systemctl status tectonic` output is fine as well. What we do is the following:

- Once the `tectonic` service is done, it write a `/opt/tectonic/init_tectonic.done` file on to disk. This makes sure that it is not run again after reboot.
- When the `tectonic` service restarts, it checks if the file is present (`ConditionPathExists=!/opt/tectonic/init_tectonic.done`). If it is, it exits without doing anything (`inactive`).

So far I can't really spot any issues on the machines. Seems like the installer GUI simply doesn't report the success of the installation. Were you able to contact the cluster via the `kubeconfig` in the assets bundle?

**Comment by megastallman** [comment]
As for the events, happened before running "# systemctl start tectonic":
No, I couldn't contact with the cluster in any ways, and had those API connection errors, described in the first message.

**Comment by knweiss** [comment]
@megastallman Did you check the output of `sudo systemctl list-jobs` and `sudo systemctl --failed` on node1 for problems that prevent `tectonic.service` from starting?

In my case it was (among others) https://github.com/systemd/systemd/issues/3752 i.e. `systemd-networkd-wait-online.service` got stuck.

**Comment by megastallman** [comment]
I've tried running `systemctl --failed` today. There was a `tectonic.service` occurrence.
[failed-units.log](https://github.com/coreos/tectonic-installer/files/1505789/failed-units.log)
